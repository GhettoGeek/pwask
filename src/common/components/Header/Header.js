import React from 'react'
import PropTypes from 'prop-types'
import { AppBar, Toolbar } from '@material-ui/core'
import { styled } from '../../utils/style'
import SideMenu from '../SideMenu'
import Typography from '../Typography'

const Wrapper = styled('div')({
  display: 'flex',
})

const Title = styled(Typography)((theme) => ({
  flexGrow: 1,
  marginLeft: theme.spacing(2),
}))

const Header = ({ title, sideMenuItems, children }) => (
  <Wrapper id="header">
    <AppBar position="static" color="primary">
      <Toolbar>
        <SideMenu items={sideMenuItems} />
        <Title
          component="h1"
          variant="h6"
          color="inherit"
          noWrap
        >
          {title}
        </Title>
        {children}
      </Toolbar>
    </AppBar>
  </Wrapper>
)

Header.propTypes = {
  title: PropTypes.string,
  sideMenuItems: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string,
    icon: PropTypes.node,
    link: PropTypes.string,
  })),
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.string]),
}

Header.defaultProps = {
  title: '',
  sideMenuItems: [],
  children: '',
}

export default Header
