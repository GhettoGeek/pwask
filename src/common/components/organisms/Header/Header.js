import React from 'react'
import PropTypes from 'prop-types'
import { withTranslation } from 'react-i18next'
import { AppBar, Toolbar, Typography } from '@material-ui/core'
import { styled } from '../../../utils/style'

import SideMenu from '../SideMenu/SideMenu'
import UserAccount from '../../molecules/UserAccount/UserAccount'

const Wrapper = styled('div')({
  display: 'flex',
})

const Title = styled(Typography)((theme) => ({
  flexGrow: 1,
  marginLeft: theme.spacing(2),
}))

const Header = ({ t }) => (
  <Wrapper id="header">
    <AppBar position="static" color="primary">
      <Toolbar>
        <SideMenu />
        <Title
          component="h1"
          variant="h6"
          color="inherit"
          noWrap
        >
          {t('header.title')}
        </Title>
        <UserAccount />
      </Toolbar>
    </AppBar>
  </Wrapper>
)

Header.propTypes = {
  t: PropTypes.func.isRequired,
}

export default withTranslation()(Header)
