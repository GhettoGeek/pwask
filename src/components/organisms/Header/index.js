import React from 'react'
import PropTypes from 'prop-types'
import { withTranslation } from 'react-i18next'
import AppBar from '@material-ui/core/AppBar'
import Toolbar from '@material-ui/core/Toolbar'
import Typography from '@material-ui/core/Typography'
import { styled } from '../../../utils/style'

import SideMenu from '../SideMenu'
import UserAccount from '../../molecules/UserAccount'

const Wrapper = styled('div')({
  display: 'flex',
})

const Title = styled(Typography)(theme => ({
  flexGrow: 1,
  marginLeft: theme.spacing.m,
}))

const Header = ({ t }) => (
  <Wrapper id="header">
    <AppBar position="static" color="primary">
      <Toolbar>
        <SideMenu />
        <Title
          component="h1"
          variant="h6"
          color="inherit"
          noWrap
        >
          {t('header.title')}
        </Title>
        <UserAccount />
      </Toolbar>
    </AppBar>
  </Wrapper>
)

Header.propTypes = {
  t: PropTypes.func.isRequired,
}

export default withTranslation()(Header)
