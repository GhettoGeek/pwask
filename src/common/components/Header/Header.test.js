import React from 'react'
import { shallow } from 'enzyme'
import Header from './Header'

describe('Header', () => {
  it('renders correctly the component', () => {
    const wrapper = shallow(<Header />)

    expect(wrapper.html()).toMatchSnapshot()
  })
})
