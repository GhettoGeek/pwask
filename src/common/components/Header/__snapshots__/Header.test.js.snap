// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly the component 1`] = `"<div class=\\"StyledComponent-root-1\\" id=\\"header\\"><header class=\\"MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary\\"><div class=\\"MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters\\"><div><button class=\\"MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-42 MuiIconButton-colorInherit\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"icon-button\\"><span class=\\"MuiIconButton-label\\"><span class=\\"material-icons MuiIcon-root\\" aria-hidden=\\"true\\" aria-label=\\"sideMenu.openDrawer\\">menu</span></span></button></div><h1 class=\\"MuiTypography-root StyledComponent-root-76 MuiTypography-h6 MuiTypography-colorInherit MuiTypography-noWrap\\"></h1></div></header></div>"`;
